In [3]:
import pandas as pd
import scipy
import numpy as np
from scipy import stats

# Cutlet Diameters Problem

In [4]:
df=pd.read_csv('Cutlets.csv')

In [5]:
df.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

Its 2 tailed test 2 sample test H0 -The daimeter of cutlets in Unit A and Unit B are same (Mu1=Mu2) 
H1-The diameter of cutlets in unit A and Unit B are different (Mu1!=Mu2777n ))

In [6]:
pvalue_tvalue = scipy.stats.ttest_1samp(df,0)
pvalue_tvalue

Ttest_1sampResult(statistic=array([143.98156495, 119.98037244]), pvalue=array([5.94226919e-49, 2.89380230e-46]))


p value is less than 0.05(significance level) so reject null hypothesis There is significance difference in diameters of the cutlets in Unit A and Unit B

# Customer Order Form Problem

In [7]:
df1=pd.read_csv('Costomer+OrderForm.csv')
df1.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

We will use chi-square test as we want to know whether the defective% in 4 telecall centers varies Meaning if there is relationship of defective % in 4 telecell centers .

In [8]:
df1['Phillippines'].replace({'Error Free':1,'Defective':0},inplace=True)
df1['Indonesia'].replace({'Error Free':1,'Defective':0},inplace=True)
df1['Malta'].replace({'Error Free':1,'Defective':0},inplace=True)
df1['India'].replace({'Error Free':1,'Defective':0},inplace=True)
df1

Phillippines  Indonesia  Malta  India
0               1          1      0      1
1               1          1      1      0
2               1          0      0      1
3               1          1      1      1
4               1          1      0      1
..            ...        ...    ...    ...
295             1          1      1      1
296             1          1      1      1
297             1          1      0      1
298             1          1      1      1
299             1          0      0      1

[300 rows x 4 columns]

In [9]:
chi_square_value, p_value, dof, exp_frq = stats.chi2_contingency(df1.values)
print("Chi-Squared Value ={} and p-value is {}".format(chi_square_value,p_value,dof))

Chi-Squared Value =112.74730231885849 and p-value is 1.0


# Lab Tat dataset problem

In [10]:
df2=pd.read_csv('LabTAT.csv')
df2.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [11]:
import scipy.stats as stats
stats.f_oneway(df2.iloc[:,0],df2.iloc[:,1],df2.iloc[:,2],df2.iloc[:,3])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

As the p value is less than significance level we reject the null hypothesis Meaning ,there is difference in the average TAT of different laboratories 

# BuyerRatio Problem

In [12]:
df3=pd.read_csv('BuyerRatio.csv')
df3.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750


h0 : Male-female ratio is equal across all regions ha : Male-female ratio is not equal accross regions

As we can see that we have categorical data in the dataset, we will do the chisq.test() to see if there is correlation between the columns .

In [18]:
df3.rename(columns={'Observed Values':'OV'},inplace=True)

In [19]:
df3.drop('OV',axis=1,inplace=True)

In [20]:
from scipy.stats import chi2_contingency

In [21]:
chi_square_value, p_value, df3, exp_frq = chi2_contingency(df3)
print("Chi-Squared Value = {} and p-value is {}".format(chi_square_value,p_value))

Chi-Squared Value = 1.595945538661058 and p-value is 0.6603094907091882


The x-squared value is 1.5959 and the p-value is 0.6603 which is greater than the 5% Significance level so we do not reject null hypothesis.we conclude that the Male-Female buyer Ratio's are similar accross regions